### ISSUE TITLE
Pattern matching on &str false positive

### ISSUE THREAD

**Issue by ice1000** [intellij-rust/intellij-rust#3791]
## Environment

Same as #3776

## Problem description
Given this minimum-reproducible-example:

```rust
enum Bla<'a> {
    Rua(&'a str)
}
fn van(bla: &Bla) {
    match bla {
        Bla::Rua("shirasawa") => (),
        Bla::Rua("marisa") => (),
        _ => unreachable!()
    }
}
```

This code compiles fine, but IntelliJ-Rust complains with:
![Screenshot from 2019-05-05 00-00-03](https://user-images.githubusercontent.com/16398479/57188207-c8a0b280-6ec8-11e9-94c4-b7a1d87dff32.png)

## Steps to reproduce

Open any project and put the code given to any file included in the project

**Comment by ice1000** [comment]
Relevant: https://github.com/owo-lang/voile-rs/commit/84e29bb3f53bc08ad9640e3d7eba944662f342ed

Sorry for saying your awesome project is "stupid" but that bug really confused me a lot.

**Comment by vlad20012** [comment]
Try to fix it today
